File: individual_modules/introduction_to_hpc/calc-squares.R
# Initialise a vector with numbers 1 to 1000
numbers <- 1:1000

# Print the initial vector
cat("Initial list of numbers:\n")
print(numbers)

# Calculate the square of each number in the vector
squares <- numeric(length(numbers))
for (i in 1:length(numbers)) {
  squares[i] <- numbers[i]^2
}

# Print the results in full
cat("\nSquares of the numbers:\n")
print(squares)

File: individual_modules/introduction_to_hpc/calc-squares.py
# Initialise a list with numbers 1 to 1000
numbers = list(range(1, 1001))

# Print the initial list
print("Initial list of numbers:")
print(numbers)

# Calculate the square of each number in the list
squares = []
for num in numbers:
    square = num ** 2
    squares.append(square)

# Print the results in full
print("\nSquares of the numbers:")
print(squares)